### PR TITLE
vagrant: log stderr when vagrant shell command fails

### PR DESCRIPTION
**Pull request by tejasmanohar**
Continuation of #536 (inlined description below) but PR'd to `master` instead of `paper` and moved log lines to callsite as proposed.
>With logs, we can (try to) figure out why the Vagrant provider is "unavailable", "unable to boot", etc. without adding one-off debugging code and/or running commands outside of Quilt.

cc @kklin @ejj

### CODE REVIEW

File: cluster/provider/vagrant_api.go [hunk 1]
```diff
@@ -63,34 +65,38 @@ func (api vagrantAPI) Init(cloudConfig string, size string, id string) error {
 }
 
 func (api vagrantAPI) Up(id string) error {
-	_, _, err := api.Shell(id, `vagrant --machine-readable up`)
+	_, stderr, err := api.Shell(id, `vagrant --machine-readable up`)
 	if err != nil {
+		log.Errorf("Failed to start Vagrant machine: %s", string(stderr))
 		return errors.New("unable to check machine status")
```
Comment (kklin) [inline cluster/provider/vagrant_api.go]: Huh, the error message I used wasn't very fitting here. Could you fix it as part of this PR?

Comment (tejasmanohar) [inline cluster/provider/vagrant_api.go]: agreed. I was a bit confused there. will do.

Comment (kklin) [inline cluster/provider/vagrant_api.go]: Thanks!

**Comment by tejasmanohar** [comment]
@kklin Fixed.

**Comment by ejj** [comment]
Overall series looks good.  Three super tiny nits.  Fix them and I'll merge.

**Comment by tejasmanohar** [comment]
@ejj fixed nits

**Pull request merged**